packages/providers/src/transaction-response/utils.ts

```diff
@@ -0,0 +1,36 @@
+import { arrayify } from '@ethersproject/bytes';
+import type { BN } from '@fuel-ts/math';
+import { ReceiptType, ReceiptCoder } from '@fuel-ts/transactions';
+
+import type { GqlReceiptFragmentFragment, GqlTransaction } from '../__generated__/operations';
+
+import type { TransactionResultReceipt } from './transaction-response';
+
+export const processGqlReceipt = (
```

Comment:
this function should be incorporate inside transaction-summary too? I think it makes sense